release/internal/version/version.go

```diff
@@ -123,30 +142,38 @@ func GitVersion() Version {
 }
 
 // HasDevTag returns true if the version has the given dev tag suffix.
-// The dev tag suffix is expected to be in the format "vX.Y.Z-<devTagSuffix>-N-gCOMMIT" or "vX.Y.Z-<devTagSuffix>-N-gCOMMIT-dirty".
+// The dev tag suffix is expected to be in one of the following formats:
+//   - vX.Y.Z-<devTagSuffix>-N-gCOMMIT
+//   - vX.Y.Z-<devTagSuffix>-N-gCOMMIT-dirty
+//   - vX.Y.Z-A.B-<devTagSuffix>-N-gCOMMIT
+//   - vX.Y.Z-A.B-<devTagSuffix>-N-gCOMMIT-dirty
+//
+// where vX.Y.Z is the semver version, <devTagSuffix> is the dev tag suffix, N is the number of commits since the tag,
+// A.B is the EP version, and COMMIT is the git commit hash abbreviated to 12 characters (e.g., 1a2b3c4d5e67).
+// The "dirty" suffix indicates that the working directory is dirty.
 func HasDevTag(v Version, devTagSuffix string) bool {
 	devTagSuffix = strings.TrimPrefix(devTagSuffix, "-")
-	re := regexp.MustCompile(fmt.Sprintf(`^v\d+\.\d+\.\d+-%s-\d+-g[0-9a-f]{12}(-dirty)?$`, devTagSuffix))
+	re := regexp.MustCompile(fmt.Sprintf(`^v\d+\.\d+\.\d+(-\d+\.\d+)?-%s-\d+-g[0-9a-f]{12}(-dirty)?$`, devTagSuffix))
 	return re.MatchString(string(v))
 }
 
 // DetermineReleaseVersion uses historical clues to figure out the next semver
 // release number to use for this release based on the current git revision.
-//
-// OSS Calico uses the following rules:
```

Comment:
I think there is value in separating the version rules that OSS uses vs enterprise?

Reply:
At the moment, the way the rules are structured apply with OSS and Enterprise although Enterprise is only the first scenario since it creates a new dev tag after every release.